contrib/archivetest.c

```diff
@@ -201,7 +201,6 @@ int main(int argc, char *argv[])
 			v_print(v, "skipping");
 		} else {
 			while ((r = archive_read_data(a, buffer, 4096) > 0))
-			;
 			if (r == ARCHIVE_FATAL) {
```

Comment:
If you fix the expression above, then your other change here has no effect, since ARCHIVE_FATAL is less than zero.